rubin_scheduler/scheduler/utils/utils.py

```diff
@@ -153,50 +154,97 @@ def mean_azimuth(az, min_val=0.1):
 class IntRounded:
     """
     Class to help force comparisons be made on scaled up integers,
-    preventing machine precision issues cross-platforms
+    preventing machine precision issues cross-platforms.
+
+    Danger zone would be doing comparisons with differences between
+    `scale` and `limit`, e.g.:
+    (1 + 1e-9) == (1 + 1e-10)
+    returns False as expected, but
+    IntRounded(1 + 1e-9) == intRounded(1 + 1e-10)
+    will return True because it rounds off at 9 decimal places.
 
     Note that casting a NaN to an int can have different behaviours
     cross-platforms, so will throw an error if attempted.
 
     Parameters
     ----------
     inval : number-like thing
-        Some number that we want to compare
-    scale : float (1e5)
+        Some number/array that we will want to compare
+    scale : `float`
         How much to scale inval before rounding and converting to an int.
+        Essentially how many digits of precesion to use in comparisons.
+        Default 1e6.
+    limit : `float`
+        If values have a differce less than the limit, force comparisions
+        to be done with scaled ints. Should be larger than whatever you
+        think the machine floating point precision might be.
+        Default 1e-9.
```

Comment:
Why not use numpy.finfo(float) (or whatever type you start with) to get this automatically?

Reply:
1) I'm not sure I believe `numpy.finfo` because it gives me the same value on systems where float comparisons behave differently (I think)
2) We need the default limit to be above the limit on the lower resolution machine. So auto setting using  `numpy.finfo` means we could be comparing native values that would be rounded to lower precision on different hardware.